src/custom/state/orders/middleware.ts

```diff
@@ -29,6 +30,14 @@ const isFulfillOrderAction = isAnyOf(OrderActions.addPendingOrder, OrderActions.
 const isExpireOrdersAction = isAnyOf(OrderActions.expireOrdersBatch, OrderActions.expireOrder)
 const isCancelOrderAction = isAnyOf(OrderActions.cancelOrder, OrderActions.cancelOrdersBatch)
 
+function reportAnalytics(action: string, label?: string) {
+  ReactGA.event({
+    category: 'Swap',
```

Comment:
wouldn't it be better to allow the category to be set? can just default to 'Swap' but we have 'Wrap' and other categories, right?

Reply:
I do these functions locally, just so its less verbose and to not keep it DRY. Its always 'Swap' here. 
In this file theres not analytics events with `Wrap` as category

Reply:
True, we could do that, and it would be a good option if we do it in all the code. This way, we abstract the implementation of the Analytics library, however this would be a bigger change cause we would need to increase the number of Mods


I still think is a good idea, I also thought about that, but discarded for this reason

Happy to do it in the future though!

Reply:
Adding here my future re-iterations including this

https://github.com/cowprotocol/cowswap/issues/665